runtime/amd64.S

```diff
@@ -233,7 +223,9 @@
 #define RESTORE_EXN_HANDLER_OCAML              \
```

Comment:
Out of curiosity: the documentation comment above say that `%r10` is clobbered, this is a small mistake, right?

Reply:
Right. I'll amend the comment.